py/dml/crep.py

```diff
@@ -19,8 +19,52 @@
     'ctype',
     'conf_obj',
     'cloggroup',
+    'dev',
+    'require_dev',
+    'maybe_dev',
+    'maybe_dev_arg',
+    'StaticContext',
+    'TypedParamContext',
     )
 
+
+class StaticContext:
+    active = False
+
+    def __enter__(self):
+        self.prev_context = StaticContext.active
+        StaticContext.active = True
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        StaticContext.active = self.prev_context
+
+class TypedParamContext(StaticContext):
+    active = False
+
+    def __enter__(self):
+        StaticContext.__enter__(self)
+        self.prev_typed_context = TypedParamContext.active
+        TypedParamContext.active = True
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        StaticContext.__exit__(self, exc_type, exc_val, exc_tb)
```

Comment:
nitpick: `__enter__` and `__exit__` have return values so `return super().__exit__(...)` is idiomatic (would have made a difference if the superclass was further away than ten lines, but it's not so ok to ignore)